h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = element.querySelector('.js-search-bar-input');
+    this._dropdown = element.querySelector('.js-search-bar-dropdown');
+    this._dropdownItems = element.querySelectorAll('.js-search-bar-dropdown-item');
+
+    var self = this;
+    var index = -1;
+    var menuLen = this._dropdownItems.length;
+
+    var updateActiveDropdownItem = function (index) {
+      clearActiveDropdownItem();
+      self._dropdownItems[index].classList.add('js-search-bar-dropdown-menu-item--active');
+    };
+
+    var clearActiveDropdownItem = function () {
+      if (element.querySelector('.js-search-bar-dropdown-menu-item--active')) {
+        element.querySelector('.js-search-bar-dropdown-menu-item--active').
+        classList.remove('js-search-bar-dropdown-menu-item--active');
+      }
+    };
+
+    var onOpenDropdown = function () {
+      self.setState({open: true});
+      self._input.addEventListener('keydown', setupListenerKeys,
+        true /*capture*/);
+    };
+
+    var onCloseDropdown = function () {
```

Comment:
In new (ES2015) code we should be consistent about using arrow functions rather than function expressions., ie. `var onCloseDropdown = () => { ... }`. This avoids the need to use `self` because the value of `this` is captured from the outer scope.